Ad9144 driver improvements

zeldin opened the pull request:
Some improvements to the ad9144 driver:
* Support AD9154
* Handle `adi,jesd-link-mode` and `adi,jesd-subclass` device tree properties

dbogdan commented:
Thanks for your patches.
First, can you please fix the checkpatch complaints: https://travis-ci.org/analogdevicesinc/linux/jobs/602889071?utm_medium=notification&utm_source=github_status ?

zeldin commented:
@dbogdan I fixed the whitespace issues, and added the signed-off-by lines.  The lack of documentation for `cf_axi_dds` isn't my fault though - it's always been undocumented...

mhennerich commented:
Nevermind! Are things now fully working, all lanes in DATA and DACs enabled?

zeldin commented:
@mhennerich Well, I haven't tested with all DAC:s yet, but AFAICT DAC selection is the same on AD9144 and AD9154, so it should work.

zeldin commented:
But yes, all status is green (also with subclass 1) and there is (correct) RF output.

mhennerich commented:
Great! We always appreciate patches and feedback.

commodo commented:
If no objections, will merge this in 2 days max
